src/main/java/io/appium/java_client/AppiumBy.java

```diff
@@ -53,6 +56,13 @@ protected AppiumBy(String selector, String locatorString, String locatorName) {
         return String.format("%s.%s: %s", AppiumBy.class.getSimpleName(), locatorName, remoteParameters.value());
     }
 
+    public Map<String, Object> toJson() {
```

Comment:
not sure this method is needed

Reply:
I can observe that Remotable.Parameters class already contains `toJson` method. It's private though. Lets add it to the list in https://github.com/SeleniumHQ/selenium/issues/13949 and leave a TODO there.

Also consider using Map.of() factory instead.

Reply:
Could you also provide an example of any of the above calls where toJson method may be used?

Reply:
One such example is using `flutter: scrollTillVisible` method.

1. **Without toJson()**

```
driver.executeScript("flutter: scrollTillVisible", Map.of("finder", Map.of("using", "-flutter key", "value", "login_button")));
```

2. **With toJson()**
```
driver.executeScript("flutter: scrollTillVisible", Map.of("finder", AppiumBy.flutterKey("login_button").toJson());
```

Reply:
It would probably make sense to create FlutterIosDriver and FlutterAndroidDriver inherited from iOSDriver and AndroidDriver, so the above extensions may be implemented as first-class methods, e.g.

flutter. scrollTillVisible(AppiumBy.flutterKey("login_button"))

Reply:
We initially had a similar idea. After reviewing the implementation, we found that the logic is consistent for both iOS and Android drivers, so we decided to maintain this approach.

Reply:
@sudharsan-selvaraj Agree we wanted to make it consistent but nothing is stopping us implementing the initial approach if everyone is fine. It gets easier to implement the tests.
cc: @mykola-mokhnach

Reply:
@saikrishna321 @mykola-mokhnach  That sounds good! Let's proceed by creating separate Flutter-specific driver classes to abstract the underlying commands. I'll prepare a patch with the changes and tag for the review.